pkg/arvo/tests/sys/hoon/list.hoon

```diff
@@ -0,0 +1,289 @@
+      !>  %.y
+      !>  (lien (limo ~[2 3]) |=([x=@] (gth 10 x)))
+  ==
+::
+++  test-list-murn  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  (limo ~[6 10])
+      !>  
+      %+  murn  (limo ~[2 3 4 5]) 
+          |=  [x=@] 
+          ^-  (unit)
+          ?:  =((mod x 2) 0)
+            [~]
+          (some (mul x 2))
+  ==
+::
+++  test-list-oust  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  (limo ~[2 5])
+      !>  (oust [1 2] (limo ~[2 3 4 5]))
+    %+  expect-eq
+      !>  (limo ~[2 3 4 5])
+      !>  (oust [1 0] (limo ~[2 3 4 5]))
+  ==
+::
+++  test-list-reap  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  (limo ~[1 1 1])
+      !>  (reap 3 1)
+  ==
+::
+++  test-list-reel  ^-  tang
+  ;:  weld
+  :: Use non-associtive operation
+    %+  expect-eq
+      !>  93
+      !>  
+      %+  reel  (limo ~[100 10 5 2]) 
+          |=  [acc=@ x=@] 
+          (sub acc x)
+  ==
+::
+++  test-list-roll  ^-  tang
+  ;:  weld
+  :: Use non-associtive operation
+    %+  expect-eq
+      !>  6
+      !>  
+      %+  roll  (limo ~[1 5 10]) 
+          |=  [acc=@ x=@] 
+          ?~  acc  x
+          (sub acc x)
+  ==
+::
+++  test-list-scag  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  (limo ~[1 2])
+      !>  (scag 2 `(list @)`(limo ~[1 2 3]))
+  ==
+::
+++  test-list-skid  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  [p=(limo ~[2 4 6 8]) q=(limo ~[3 5 7])]
+      !>  %+  skid  (limo ~[2 3 4 5 6 7 8])
+          |=  [x=@]
+          =((mod x 2) 0)
+  ==
+::
+++  test-list-skim  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  (limo ~[2 4 6 8])
+      !>  %+  skim  (limo ~[2 3 4 5 6 7 8])
+          |=  [x=@]
+          =((mod x 2) 0)
+  ==
+::
+++  test-list-skip  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  (limo ~[3 5 7])
+      !>  %+  skip  (limo ~[2 3 4 5 6 7 8])
+          |=  [x=@]
+          =((mod x 2) 0)
+  ==
+::
+++  test-list-slag  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  (limo ~[3])
+      !>  (slag 2 (limo ~[1 2 3]))
+  ==
+::
+++  test-list-snag  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  3
+      !>  (snag 2 (limo ~[1 2 3]))
+  ==
+::
+++  test-list-sort  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  (limo ~[1 2 3])
+      !>  (sort (limo ~[2 1 3]) lth)
+    %+  expect-eq
+      !>  (sort (limo ~[2 1 3]) lth) :: Todo add random lists here
+      !>  (sort (sort (limo ~[2 1 3]) lth) lth)
+  ==
+::
+++  test-list-spin  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  [p=`(list @)`(limo ~[0 3 8]) q=3]
+      !>  %^  spin  (limo ~[2 3 4])  0
+          |=  [x=@ y=@]
+          [(mul x y) +(y)]
+    ==
+::
+++  test-list-spun  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  `(list @)`(limo ~[0 3 8])
+      !>  %+  spun  (limo ~[2 3 4])
+          |=  [x=@ y=@]
+          [(mul x y) +(y)]
+    ==
+::
+++  test-list-turn  ^-  tang
+  ;:  weld
+    %+  expect-eq
+      !>  `(list @)`(limo ~[3 4 5])
+      !>  %+  turn  (limo ~[2 3 4])
+          |=  [x=@]
+          +(x)
```

Comment:
Brackets are unnecessary when constructing a mold of one item. And this would probably be better on one line: `|=(x=@ +(x))`.